yarn-project/stdlib/src/abi/contract_artifact.ts

```diff
@@ -215,18 +216,31 @@ function generateFunctionArtifact(
 }
 
 function getFunctionType(fn: NoirCompiledContractFunction): FunctionType {
-  if (fn.custom_attributes.includes(AZTEC_PRIVATE_ATTRIBUTE)) {
+  if (fn.custom_attributes.some(attr => attr.endsWith(AZTEC_PRIVATE_ATTRIBUTE))) {
     return FunctionType.PRIVATE;
-  } else if (fn.custom_attributes.includes(AZTEC_PUBLIC_ATTRIBUTE)) {
+  } else if (fn.custom_attributes.some(attr => attr.endsWith(AZTEC_PUBLIC_ATTRIBUTE))) {
     return FunctionType.PUBLIC;
-  } else if (fn.is_unconstrained) {
-    return FunctionType.UNCONSTRAINED;
+  } else if (fn.custom_attributes.some(attr => attr.endsWith(AZTEC_UTILITY_ATTRIBUTE))) {
+    return FunctionType.UTILITY;
   } else {
-    // Default to a private function (see simple_macro_example_expanded for an example of this behavior)
-    return FunctionType.PRIVATE;
+    throw new Error(`Invalid function type for a noir contract function ${fn.name}`);
```

Comment:
Defaulting to PRIVATE is the kind of thing that results in a brittle hard-to-debug code so I am erroring here now instead.